### ISSUE TITLE
Lookahead fst libraries are not included in path

### ISSUE THREAD

**Issue by Malkovsky** [kaldi-asr/kaldi#3959]
Recipes for recently added lookahead decoding fail because lookahead libraries are built in `<openfst>/lib/fst` rather then `<openfst>/lib`

adding
`export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$KALDI_ROOT/tools/openfst/lib/fst`
to path.sh solves the problem

**Comment by nshmyrev** [comment]
I'm sorry, the LD_LIBRARY_PATH is already exported in 

https://github.com/kaldi-asr/kaldi/blob/88e7b85e2cc0c4822114850cb851d36e9ed5bc5a/egs/mini_librispeech/s5/local/lookahead/run_lookahead.sh#L42

I don't think you need to modify it for every project.

**Comment by Malkovsky** [comment]
Oh, i see i missed that one. I would still suggest that it belongs to `tools/config/common_path.sh` and the current export in run_lookahead.sh overwrites LD_LIBRARY_PATH rather then appends to it and thus might cause problems

**Comment by danpovey** [comment]
In general we use -rpath to bake those kinds of paths into the binaries.  I
think that would be a better solution-- would require changes to the
configure script or maybe the makefiles/*.

**Comment by nshmyrev** [comment]
rpath wouldn't work since this .so plugin is loaded by fst tools too, or rpath has to be configured in openfst in openfst too. Let me know if you want to keep it as is or modify common_path.sh please.

**Comment by danpovey** [comment]
er... I'm a little surprised by that because generally OpenFst tools work
"out of the box" with no need for special paths, but maybe it's because
it's an extension.  Modifying common_path.sh is OK, if that's the path of
lease resistance.   Make sure it does the right thing in case it's run
twice.

**Comment by danpovey** [comment]
@nshmyrev I see you have a fix but I don't know of a PR.

**Comment by nshmyrev** [comment]
@danpovey sorry, no pr yet. I'll implement online decoder this week as requested and submit.

**Comment by Hap-Zhang** [comment]
Hi,all
i want to load a Hclr.fst by the code "hcl_fst = fst::StdFst::Read(hcl_fst_rxfilename)", and get a error as "ERROR: GenericRegister::GetEntry: lookup failed in shared object: olabel_lookahead-fst.so", would you like to tell me any mistake when build openfst?

**Comment by danpovey** [comment]
I don't know about the specific error, but how recently did you install
OpenFST?  Because we only added the option --enable-lookahead-fsts to our
Makefile in tools/ fairly recently, I think.

**Comment by Hap-Zhang** [comment]
Yes,i config openfst-1.6.1 with option --enable-lookahead-fsts and build&install recently.I want to try lookahead graph decoding when online decoding,and unluckily i get this error.

**Comment by danpovey** [comment]
Please ask on the OpenFST forums, since I don't know enough to solve this.
If they don't respond in a day or two you can ask me again and I'll ask
someone who might know.

**Comment by Hap-Zhang** [comment]
ok,i'll have a try on the OpenFST forums firstly,thank you.

**Comment by Malkovsky** [comment]
@Hap-Zhang This is most likely due to the issue of this thread: you have to make sure that <openfst>/tools/openfst/lib/fst is in your $LD_LIBRARY_PATH variable which is exported in the run_lookahead.sh as mentioned @nshmyrev, but if you want to do some experiments outside of that script, you need to export paths like
`export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$KALDI_ROOT/tools/openfst/lib/fst`

**Comment by Hap-Zhang** [comment]
@Malkovsky Yeah,i see. I really export paths like:`export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$KALDI_ROOT/tools/openfst/lib/fst`
However, the error is:"lookup failed in shared object: olabel_lookahead-fst.so", and not "olabel_lookahead-fst.so: cannot open shared object file: No such file or directory".
So, maybe any other problem？

**Comment by Malkovsky** [comment]
@Hap-Zhang Hmm, indeed, i didn't encounter such a problem

**Comment by nshmyrev** [comment]
@Hap-Zhang First of all you need to check if the file is actually present in your filesystem. It should be in $KALDI_ROOT/tools/openfst/lib/fst folder.

Second, the demo and kaldi works with openfst-1.6.7 by default. I am not sure why do you try with 1.6.1

**Comment by rado-r** [comment]
hey @nshmyrev maybe there typo in your commit is `export LD_LIRBRARY_PATH=$LD_LIBRARY_PATH:${KALDI_ROOT}/tools/openfst/lib/fst`
and there should be `LD_LIBRARY_PATH ` instead of `LD_LIRBRARY_PATH`

**Comment by nshmyrev** [comment]
> hey @nshmyrev maybe there typo in your commit is export 

Great catch! Please submit a pull request for the fix
